### PR TITLE
Vision AutoML samples

### PR DESCRIPTION
**Pull request by nirupa-kumar**
Fixes #<issue_number_goes_here> (it's a good idea to open an issue first for discussion)

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,10 +65,12 @@
     "test": "npm run cover"
   },
   "dependencies": {
+    "@google-cloud/automl": "^0.1.2",
```
Comment (JustinBeckwith) [inline package.json]: I don't think we want to have vision depend on automl, do we?

File: package.json [hunk 2]
```diff
@@ -65,10 +65,12 @@
...
     "@google-cloud/promisify": "^0.3.0",
     "google-gax": "^0.20.0",
     "is": "^3.2.1",
     "lodash.merge": "^4.6.1",
+    "mathjs": "^5.1.2",
```
Comment (JustinBeckwith) [inline package.json]: Guessing this was supposed to be in the samples package.json

File: samples/automl/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "automl",
+  "version": "1.0.0",
+  "description": "",
+  "main": "automlVisionDataset.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (JustinBeckwith) [inline samples/automl/package.json]: Would be great to be able to run the tests directly from here.

File: samples/package.json [hunk 4]
```diff
@@ -12,16 +12,20 @@
...
     "async": "^2.6.1",
+    "mathjs": "^5.0.4",
     "natural": "^0.6.1",
     "redis": "^2.8.0",
     "yargs": "^12.0.0"
   },
   "devDependencies": {
     "@google-cloud/nodejs-repo-tools": "^2.3.0",
-    "@google-cloud/storage": "^2.0.0",
+    "@google-cloud/storage": "^1.7.0",
```
Comment (JustinBeckwith) [inline samples/package.json]: we need to bump this to 2.0 now

File: samples/system-test/automlVision.test.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require(`path`);
+//const storage = require(`@google-cloud/storage`)();
```
Comment (JustinBeckwith) [inline samples/system-test/automlVision.test.js]: why commented out?

**Comment by JustinBeckwith** [comment]
Making sure @beccasaurus has seen this :)

File: package.json (hunk 1 above)
Comment (nirupa-kumar) [reply package.json]: Fixed in latest commit.

File: package.json (hunk 2 above)
Comment (nirupa-kumar) [reply package.json]: Fixed in latest commit.

File: samples/automl/package.json (hunk 3 above)
Comment (nirupa-kumar) [reply samples/automl/package.json]: Fixed in latest commit.

File: samples/package.json (hunk 4 above)
Comment (nirupa-kumar) [reply samples/package.json]: Fixed in latest commit.

File: samples/system-test/automlVision.test.js (hunk 5 above)
Comment (nirupa-kumar) [reply samples/system-test/automlVision.test.js]: None of the tests seemed to need GCS. Will look into this to make sure.

**Comment by nirupa-kumar** [comment]
@beccasaurus  Please have a look. There is one pending issue raised by Justin that I'm currently looking into.

File: samples/automl/automlVisionDataset.js [hunk 6]
```diff
@@ -0,0 +1,404 @@
+/**
+ * Copyright 2018, Google, Inc.
```
Comment (happyhuman) [inline samples/automl/automlVisionDataset.js]: Google, LLC

Comment (nirupa-kumar) [reply samples/automl/automlVisionDataset.js]: fixed in latest commit.

**Comment by nirupa-kumar** [comment]
@happyhuman Please let me know if this is good to merge.

**Pull request merged**